### ISSUE TITLE
Change in the Sentimental Analysis

### ISSUE THREAD

**Issue by Niranjan-guru** [BharathK05/Sentimental-Analysis-Tweet#1]
Hey Bharath, 
     The issue in the Flask's rendering process is now cleared and the code is enhanced a little bit..and I tried my best to come over it...please do contact me if it still persists...

**Html Code:-**
<!DOCTYPE html>
<html>
<body>

<form action="/process" method="post">
  <label for="user_input">Input:</label>
  <input type="text" id="user_input" name="user_input">
  <button type="submit">Submit</button>
</form>

{% if user_input %}
  <p>Input: {{ user_input }}</p>
{% endif %}

{% if output %}
  <p>Output from Python code:</p>
  <p>{{ output }}</p>
{% endif %}

</body>
</html>

**Python**
from flask import Flask, render_template, request
import subprocess

app = Flask(__name__)

@app.route('/')
def index():
    return render_template('index.html')

@app.route('/process', methods=['POST'])
def process():
    user_input = request.form['user_input']
    
    # Assuming 'tw-sentimental.py' is in the same directory as your Flask app
    python_output = subprocess.run(['python', 'tw-sentimental.py', user_input], capture_output=True, text=True).stdout
    
    return render_template('index.html', user_input=user_input, output=python_output)

if __name__ == '__main__':
    app.run(debug=True)

**Main Sentimental Analysis**
from transformers import AutoTokenizer, AutoModelForSequenceClassification
from scipy.special import softmax
import numpy as np
import sys

# Input of dataset
tweet = sys.argv[1]

# Preprocessing Data
tweet_word = []
for word in tweet.split(" "):
    if word.startswith('@') and len(word) > 1:
        word = "@user"
    elif word.startswith('http'):
        word = "http"
    tweet_word.append(word)

tweet_proc = ''.join(tweet_word)  # The Cleaned Data is stored Here

# Loading the Model
roberta = "cardiffnlp/twitter-roberta-base-sentiment"
model = AutoModelForSequenceClassification.from_pretrained(roberta)
tokenizer = AutoTokenizer.from_pretrained(roberta)

labels = ['Negative', 'Neutral', 'Positive']

# Main-Sentimental Analysis
encoded_tweet = tokenizer(tweet_proc, return_tensors='pt')

# Output after passing it to the Model
output = model(**encoded_tweet)
prob = output.logits[0].detach().numpy()
prob = softmax(prob)

# Printing the predicted result
a = [np.round(float(s), 4) for s in prob]
c = max(a)
d = a.index(c)
e = labels[d]
print(f"It is a {e} Comment, With a Probability of {c * 100}%")

Thank you,
